client/src/http_default.rs

```diff
@@ -41,90 +41,64 @@ trait SetHeadersExt: Sized + SetSingleHeader {
     }
 }
 
-impl SetHeadersExt for AttoHttpRequestBuilderWithBytes {}
+impl SetHeadersExt for AttoHttpRequestBuilder {}
 
 impl SetHeadersExt for http::request::Builder {}
 
 /// Default request builder & sender implemented on top of `attohttpc` crate.
 ///
 /// Its main goal is not to be efficient, but simple. Its implementation contains
 /// some intermediate allocations that could be avoided with additional complexity.
-pub struct DefaultRequestBuilder {
-    method: Method,
-    url: String,
-    params: Option<Vec<(String, String)>>,
-    headers: Option<Headers>,
-    body: Option<Bytes>,
-}
+pub struct DefaultRequestBuilder(Result<AttoHttpRequestBuilder>);
 
 impl DefaultRequestBuilder {
+    fn and_then<F>(self, fun: F) -> Self
```

Comment:
Forward documentation about semantics. Right now there's nothing.

Reply:
Thought it is not needed here because it's private

Reply:
Some functions need an explanation regardless. Thank you for adding the docs.